### DISCUSSION TITLE
Is it possible to use this lib with React 18?

### DISCUSSION THREAD

**Discussion by samuelgrigolato** [software-mansion/react-native-screens#1522] [Q&A] [answered]
I'm trying to update React on a project that uses react-native-screens as a dependency of react-navigation. The current version is 17 and I want to bump it to 18. The problem I'm facing is that react-native-screens depends on react-freeze (https://github.com/software-mansion/react-native-screens/blob/main/package.json#L60), which itself depends on react 17 (https://github.com/software-mansion/react-freeze/blob/main/package.json#L32). As far as I can see there's no way around this limitation for now, so my options would be 1) wait a new version of react-native-screens that works with React 18, or; 2) remove the dependency on react-navigation entirely and deal with navigation another way. Does this make sense?

PS: I don't want to enable legacy-peer-dependencies and I'm trying to avoid sticking with React 17 one more release cycle (that would be a last resort option only).

**Comment by samuelgrigolato** [comment]
> To configure react-navigation to use screens instead of plain RN Views for rendering screen views

This makes me wonder if I can use react-navigation without react-native-screens entirely, never tried it as react-navigation's docs make it seem like the dependency is mandatory (https://reactnavigation.org/docs/getting-started/).

> The libraries we will install now are [react-native-screens](https://github.com/software-mansion/react-native-screens) and [react-native-safe-area-context](https://github.com/th3rdwave/react-native-safe-area-context). If you already have these libraries installed and at the latest version, you are done here! Otherwise, read on.

**Reply by samuelgrigolato** [reply]
It seems like this is not the only issue one will be facing when trying to use React 18 with React Native right now (https://github.com/facebook/react-native/pull/34104). Maybe the bump will have to wait a bit more.

**Comment by tboba** [comment, marked as answer]
Hi @samuelgrigolato, I believe the newest version of react-native-screens (which at the moment is 3.26.0) should work correctly with React 18. If you still spot any problems related to the React version, please submit an issue with steps to reproduce (including minimal repro).

Cheers!
